packages/app/src/hooks/useLoadProjectWithFileBrowser.ts

```diff
@@ -31,7 +31,7 @@ export function useLoadProjectWithFileBrowser() {
           (p) => p.project.metadata.id === project.metadata.id,
         );
 
-        if (alreadyOpenedProject) {
+        if (alreadyOpenedProject && alreadyOpenedProject.fsPath) {
```

Comment:
I think `alreadyOpenedPath.fsPath` can be undefined, if the project is a new project that hasn't yet been saved.

Perhaps we can move the check into the `toast.error` call instead?

Reply:
If the project hasn't been saved then a user wouldn't be able to use `useLoadProjectWithFileBrowser` would they? Then it seems fine to have the check here. Not sure if I'm interpreting this right

Reply:
Additionally, the purpose of this if condition only wraps the toast, if I move the check within the `toast.error`, I'd still be dealing with the unwarranted error in loading a project
<img width="512" alt="Screenshot 2024-01-12 at 1 38 43 PM" src="https://github.com/Ironclad/rivet/assets/107009412/b08a89c6-3fc4-45cd-8267-d277c665a377">

Reply:
I think my specific suggestion was:

```typescript
if (alreadyOpenedProject) {
  const fileName = alreadyOpenedProject.fsPath?.split('/').pop() ?? '';
  toast.error(`"${alreadyOpenedProject.project.metadata.title} [${fileName}]" shares the same ID (...`);
  return;
}
```

Reply:
I guess I'm not sure how we get into this situation in the first place ever, so I'm inclined to suggest programming super defensively.

If `alreadyOpenedProject` has the same ID as the project we are about to open, we will likely run into issues where the wrong project gets updated in state. Hence why I'd worry about adding the `&& alreadyOpenedProject.fsPath`.

Reply:
Gotcha! Thanks for clarifying! That makes sense!